chore(docs): add useFragment_experimental documentation (#10099)
apollographql/apollo-client: docs/shared/useFragment-options.mdx

<table class="field-table api-ref">

<thead>
    <tr>
      <th>Name /<br/>Type</th>
      <th>Description</th>
    </tr>
  </thead>

<tbody>

<tr>
<td colspan="2">

**Operation options**

</td>
</tr>

<tr class="required">
<td>

###### `from`

`string | StoreObject | Reference`
</td>

<td>

> **Required.** The cached object (or object reference) for the fragment being retrieved.

Reviewer: ```suggestion
**Required.** An object containing a `__typename` and primary key fields (such as `id`) identifying the entity object from which the fragment will be retrieved, or a `{ __ref: "..." }` reference, or a `string` ID (uncommon)
```
Need to be a little more specific than "the cached object"

Author: Thanks, +1 for the details. I took this from the `InMemoryCache` docs [section on the `identify` method's signature](https://github.com/apollographql/apollo-client/blob/ee2ff81b7efeb4d3e90a2e2868accf1ef46553ca/docs/source/api/cache/InMemoryCache.mdx#identify) and would be happy to update it there too.

<table class="field-table api-ref">

<thead>
    <tr>
      <th>Name /<br/>Type</th>
      <th>Description</th>
    </tr>
  </thead>

<tbody>

<tr>
<td colspan="2">

**Operation options**

</td>
</tr>

> <tr>

Reviewer 2: This is awesome! To match the exact style of other "field tables":

* Add `class="required"` to each `tr` for a required field (this applies a subtle shading to the row)
* You can omit "**Optional**" for optional parameters

<td>

###### `from`

`string | StoreObject | Reference`
</td>

<td>

**Required.** The cached object (or object reference) for the fragment being retrieved.

</td>
</tr>

<tr>
<td>

###### `fragment`

`DocumentNode`
</td>

<td>

**Required.** A GraphQL fragment document parsed into an AST with the `gql` template literal.

</td>
</tr>

<tr>
<td>

###### `fragmentName`

`string`
</td>

<td>

> **Optional.** The name of the fragment defined in the `fragment` document to use in the call.

Reviewer 2: ```suggestion
The name of the fragment defined in the [`fragment`](#fragment) document to use in the call.
```

Reviewer 2: This ability for cross-parameter linking is a big part of why we use this absurd syntax for field tables 😄

Reviewer 2: Hm I wonder if the anchor will work for a shared MDX snippet... I guess we'll see!

Author: Looks like it does 😁🎉

> You don't need to provide this value if the `fragment` document includes only one fragment.

Reviewer 2: ```suggestion
**Required** if the `fragment` document includes more than one fragment, optional otherwise.
```

Reviewer 2: One of these weird parameters that's required only under certain circumstances 😄 I think I usually go for a middle ground like this where the table row isn't shaded, but we still emphasize the case where the parameter _is_ necessary ahead of when it's optional.
